test/Game/GameCheckmateTest.test.ts

```diff
@@ -9,454 +9,170 @@ import { Rook } from "../../src/Board/Rook";
+    //given
+    const kWhite = new King(PieceColor.White, PieceType.King, new Square(7, 7));
+    const rWhite = new Rook(PieceColor.White, PieceType.Rook, new Square(1, 1));
+    const kBlack = new King(PieceColor.Black, PieceType.King, new Square(0, 0));
+    const board = new Board();
+    board.addPiece(kWhite);
+    board.addPiece(rWhite);
+    board.addPiece(kBlack);
+    const game = new Game(undefined, PieceColor.White, undefined, undefined, undefined, board);
+    // when
+    const checkMateMove = new Move(new Square(1,1), new Square(1,0), rWhite, null);
+    // then
+    const checkMateResult = game.isCheckmated();
+    const checkResult = game.isChecked();
+    expect(checkMateResult).toBe(false);
+    expect(checkResult).toBe(true);
   });
 
   //TEST 4//
 
-  test("check white King checkmate on position [2][2]", () => {
-    const newBoard = new Board();
-
-    const whiteKingPosition = new Square(2, 2);
-    const whiteBishopPosition = new Square(3, 4);
-    const whitePawnPosition = new Square(7, 8);
-
-    const blackKingPosition = new Square(7, 5);
-    const blackKnightPosition = new Square(6, 1);
-    const blackRookPosition = new Square(3, 1);
-    const blackQueenPosition = new Square(5, 5);
-
-    const whiteKing = new King(
-      PieceColor.White,
-      PieceType.King,
-      whiteKingPosition
-    );
-
-    const whiteBishop = new Bishop(
-      PieceColor.White,
-      PieceType.Bishop,
-      whiteBishopPosition
-    );
-
-    const whitePawn = new Pawn(
-      PieceColor.White,
-      PieceType.Pawn,
-      whitePawnPosition
-    );
-
-    const blackKing = new King(
-      PieceColor.Black,
-      PieceType.King,
-      blackKingPosition
-    );
-
-    const blackKnight = new Knight(
-      PieceColor.Black,
-      PieceType.Knight,
-      blackKnightPosition
-    );
-
-    const blackRook = new Rook(
-      PieceColor.Black,
-      PieceType.Rook,
-      blackRookPosition
-    );
-
-    const blackQueen = new Queen(
-      PieceColor.Black,
-      PieceType.Queen,
-      blackQueenPosition
-    );
-
-    newBoard.addPiece(whiteKing);
-    newBoard.addPiece(whiteBishop);
-    newBoard.addPiece(whitePawn);
-    newBoard.addPiece(blackKing);
-    newBoard.addPiece(blackKnight);
-    newBoard.addPiece(blackRook);
-    newBoard.addPiece(blackQueen);
-
-    const newGame = new Game(
-      undefined,
-      undefined,
-      undefined,
-      undefined,
-      undefined,
-      newBoard
-    );
-
-    const result = newGame.isCheckmated();
-    expect(result).toBe(true);
+  test("BLACK KING CHECKMATE BY TWO ROOKS", () => {
+    //given
+    const kWhite = new King(PieceColor.White, PieceType.King, new Square(7, 7));
+    const r1White = new Rook(PieceColor.White, PieceType.Rook, new Square(3, 1));
+    const r2White = new Rook(PieceColor.White, PieceType.Rook, new Square(2, 2));
+    const kBlack = new King(PieceColor.Black, PieceType.King, new Square(0, 0));
```

Comment:
```suggestion
    const kingBlack = new King(PieceColor.Black, PieceType.King, new Square(0, 0));
```
byłoby lepsze, więcej mówiące